litex_boards/community/platforms/marblemini.py

```diff
@@ -0,0 +1,253 @@
+    'LA33_P': 'J20'
+}), ("LPC-2", {
+    'CLK0_M2C_N': 'W4',
+    'CLK0_M2C_P': 'V4',
+    'CLK1_M2C_N': 'T4',
+    'CLK1_M2C_P': 'R4',
+    'LA00_N': 'U5',
+    'LA00_P': 'T5',
+    'LA01_N': 'AA4',
+    'LA01_P': 'Y4',
+    'LA02_N': 'V3',
+    'LA02_P': 'U3',
+    'LA03_N': 'V2',
+    'LA03_P': 'U2',
+    'LA04_N': 'V5',
+    'LA04_P': 'U6',
+    'LA05_N': 'T6',
+    'LA05_P': 'R6',
+    'LA06_N': 'U1',
+    'LA06_P': 'T1',
+    'LA07_N': 'V8',
+    'LA07_P': 'V9',
+    'LA08_N': 'Y2',
+    'LA08_P': 'W2',
+    'LA09_N': 'AA3',
+    'LA09_P': 'Y3',
+    'LA10_N': 'Y1',
+    'LA10_P': 'W1',
+    'LA11_N': 'AA6',
+    'LA11_P': 'Y6',
+    'LA12_N': 'W5',
+    'LA12_P': 'W6',
+    'LA13_N': 'W7',
+    'LA13_P': 'V7',
+    'LA14_N': 'AB1',
+    'LA14_P': 'AA1',
+    'LA15_N': 'AB5',
+    'LA15_P': 'AA5',
+    'LA16_N': 'AB2',
+    'LA16_P': 'AB3',
+    'LA17_N': 'W12',
+    'LA17_P': 'W11',
+    'LA18_N': 'V14',
+    'LA18_P': 'V13',
+    'LA19_N': 'Y12',
+    'LA19_P': 'Y11',
+    'LA20_N': 'AA11',
+    'LA20_P': 'AA10',
+    'LA21_N': 'AA14',
+    'LA21_P': 'Y13',
+    'LA22_N': 'W16',
+    'LA22_P': 'W15',
+    'LA23_N': 'W10',
+    'LA23_P': 'V10',
+    'LA24_N': 'Y14',
+    'LA24_P': 'W14',
+    'LA25_N': 'AB12',
+    'LA25_P': 'AB11',
+    'LA26_N': 'AA16',
+    'LA26_P': 'Y16',
+    'LA27_N': 'AB10',
+    'LA27_P': 'AA9',
+    'LA28_N': 'T15',
+    'LA28_P': 'T14',
+    'LA29_N': 'U16',
+    'LA29_P': 'T16',
+    'LA30_N': 'V15',
+    'LA30_P': 'U15',
+    'LA31_N': 'AB13',
+    'LA31_P': 'AA13',
+    'LA32_N': 'AB15',
+    'LA32_P': 'AA15',
+    'LA33_N': 'AB17',
+    'LA33_P': 'AB16'
+})]
+
+# Platform -----------------------------------------------------------------------------------------
+
+
+class Platform(XilinxPlatform):
+    def __init__(self):
+        XilinxPlatform.__init__(
+            self, "xc7a100t-2fgg484", _io, _connectors, toolchain="vivado")
+        # For now set to use the default VIVADO programmer, should be changed to xc3sprog
+        self.toolchain.bitstream_commands = [
+            "set_property BITSTREAM.CONFIG.SPI_BUSWIDTH 4 [current_design]"
+        ]
+        self.toolchain.additional_commands = [
+            "write_cfgmem -force -format bin -interface spix4 -size 16 -loadbit \"up 0x0 {build_name}.bit\" -file {build_name}.bin"
+        ]
+        self.add_platform_command("set_property INTERNAL_VREF 0.675 [get_iobanks 35]")
+        self.add_platform_command("set_property CFGBVS VCCO [current_design]")
+        self.add_platform_command("set_property CONFIG_VOLTAGE 3.3 [current_design]")
+
+    def create_programmer(self):
+        # TODO: Should be changed to xc3sprog if confirmed so
+        return VivadoProgrammer()
+
+    def do_finalize(self, fragment):
+        XilinxPlatform.do_finalize(self, fragment)
+        try:
+            self.add_period_constraint(
+                self.lookup_request("clk20_vcxo"), 1e9 / 20e6)
+        except ConstraintError:
+            pass
+        try:
```

Comment:
Now you can just do: `self.add_period_constraint(self.lookup_request("clk20_vcxo", loose=True), 1e9/20e6)` (`loose=True` is there to allow the lookup_request to fail if not found, similarly to what was done before but in a single line).